### PR TITLE
feat: add pester testcases

### LINKED ISSUE #117: Add pester testcases for automated testing of the code
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Today we have manual testing process to test the VMware.CloudFoundation.CertificateManagement code, but this has to be put in CD/CI pipeline to automate the testing process for repeatable testing before pushing to public GitHub to test on the quality of the fix and to minimize regression defects. This also ensures, time consumed for less in testing the code.

### Use Case(s)

Suppose if manual testing takes about x time, automation of the same takes x/2. Also it is more reliable and you can test even negative testcases easily.

### Potential Configuration

writing pester testcases to automate the testing process
In the current code structure, there will be /tests folder, under that pester testcases will be available for testing
Also it requires input json file to hold all the credential information

### References

_No response_

### PR DESCRIPTION
**Pull request by Ashwini-Ravi**
### Summary

Adds Pester testcases.

### Type

- [ ] Bugfix
- [X] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### Test and Documentation

- [X] Tests have been completed.
- [ ] Documentation has been added or updated.

### Issue References

Closes #117

### CODE REVIEW

**Comment by bhumitra** [comment]
@Ashwini-Ravi do you have a demo/screenshot for this which show it in action?

File: tests/inputData.json.example [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+    "SDDC Manager": "sfo-vcf01.sfo.rainpole.io",
+    "SDDC Manager VM Name": "sfo-vcf01",
+    "User": "administrator@vsphere.local",
+    "Password": "VMw@re123!",
+    "certAuthorityFqdn": "rpl-dc01.rainpole.io",
+    "certAuthorityUser": "svc-vcf-ca",
+    "certAuthorityPass": "VMw@re123!",
```
Comment (garlicNova) [inline tests/inputData.json.example:8]:
```suggestion
    "certAuthorityPass": "VMware1!",
```

File: tests/inputData.json.example [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+    "SDDC Manager": "sfo-vcf01.sfo.rainpole.io",
+    "SDDC Manager VM Name": "sfo-vcf01",
+    "User": "administrator@vsphere.local",
+    "Password": "VMw@re123!",
```
Comment (garlicNova) [inline tests/inputData.json.example:5]:
```suggestion
    "Password": "VMware1!",
```

File: tests/cert.tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+        $organizationUnit = $inputData.'organizationUnit'
+        $locality = $inputData.'locality'
+        $state = $inputData.'state'
+        $country = $inputData.'country'
+        $outputDirectory = $inputData.'outputDirectory'
+        $email = $inputData.'email'
+        $keysize = $inputData.'keysize'
+        $domain = $inputData.'Domains'[0]
+        $esxiServer = $inputData.$domain.'ESXi Hosts'[0]
+        $cluster = $inputData.$domain.'Clusters'[0]
```
Comment (garlicNova) [inline tests/cert.tests.ps1:120]:
```suggestion
        #$esxiServer = $inputData.$domain.'ESXi Hosts'[0]
        #$cluster = $inputData.$domain.'Clusters'[0]
```
since these are not used should be comment out for now

File: tests/inputData.json.example [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "sfo-m01": {
+        "ESXi Hosts": [
+            "sfo01-m01-esx01.sfo.rainpole.io",
+            "sfo01-m01-esx02.sfo.rainpole.io",
+            "sfo01-m01-esx03.sfo.rainpole.io",
+            "sfo01-m01-esx04.sfo.rainpole.io"
+        ],
+        "Clusters": [
+            "sfo-m01-cl01"
+        ]
```
Comment (garlicNova) [inline tests/inputData.json.example:32]:
```suggestion
        "Clusters": {
        Name: "sfo-m01-cl01",
        "ESXi Hosts": [
            "sfo01-m01-esx01.sfo.rainpole.io",
            "sfo01-m01-esx02.sfo.rainpole.io",
            "sfo01-m01-esx03.sfo.rainpole.io",
            "sfo01-m01-esx04.sfo.rainpole.io"
        ]
        }
```
this way, the esxi hosts are listed under cluster in case there are multiple clusters

**Review by garlicNova — CHANGES_REQUESTED** [review]

File: tests/cert.tests.ps1 (hunk 3 above)
Comment (tenthirtyam) [reply tests/cert.tests.ps1:120]: Best to just remove instead of comment.

```suggestion
```

File: tests/inputData.json.example (hunk 4 above)
Comment (tenthirtyam) [reply tests/inputData.json.example]: @Ashwini-Ravi addressed this in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/commit/400b75366f04d6c47d1edc5acbd905f9219adead.

File: tests/inputData.json.example (hunk 2 above)
Comment (tenthirtyam) [reply tests/inputData.json.example]: @Ashwini-Ravi addressed this in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/commit/400b75366f04d6c47d1edc5acbd905f9219adead.

File: tests/cert.tests.ps1 (hunk 3 above)
Comment (tenthirtyam) [reply tests/cert.tests.ps1]: @Ashwini-Ravi addressed this in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/commit/400b75366f04d6c47d1edc5acbd905f9219adead.

**Review by tenthirtyam — APPROVED** [review]
LGTM! Thank you, @Ashwini-Ravi! 🎉 

@sowjuec will pick it up from here if any changes are required post-merge.
